internal/handlemap/handle_map.go

```diff
@@ -56,6 +56,8 @@ const (
 
 type Cache struct {
 	*cache_policy.LRUCache
+	*common.BlockOffsetList
+	StreamOnly bool
```

Comment:
do we need this per handle ? streamonly or cache-and-stream is a stream component level decision ?

Reply:
yes it's per handle since we want to dictate if a handle limit is reached then the handle should stream only until we have a free spot again